### PR TITLE
InfluxDB V2 HTTPS cloud access

### PR DESCRIPTION
**Pull request by cail**
Making code to work over HTTPS connection.
Server's fingerprint is required to be specified manually.
Fixed a bug where subsequent writes() gets failed due to missing 'Token' header

### CODE REVIEW

File: InfluxDb.cpp [hunk 1]
```diff
@@ -137,6 +153,8 @@ boolean Influxdb::write(String data) {
     Serial.print(" --> writing to " + _db + ":\n");
     Serial.println(data);
   }
+  if(_db_v == 2)
+    http.addHeader("Authorization", "Token " + _token);
```
Comment (tobiasschuerg) [inline InfluxDb.cpp]: Is it really necessary to add the header again as it is already added in line 104?

File: InfluxDb.cpp [hunk 2]
```diff
@@ -81,10 +81,26 @@ void Influxdb::setVersion(uint16_t version){
...
+}
+
 void Influxdb::begin() {
   // TODO: recreate HttpClient on db change?
   if(_db_v == 2){
-    http.begin(_host, _port, "/api/v2/write?org=" + _org + "&bucket=" + _bucket);
+    if (_port == 443) {
+        if (_fingerPrint)
+          client.setFingerprint(_fingerPrint);
+        http.begin(client, _host, _port, "/api/v2/write?org=" + _org + "&bucket=" + _bucket,
```
Comment (tobiasschuerg) [inline InfluxDb.cpp]: Maybe put that on one line
```suggestion
        http.begin(client, _host, _port, "/api/v2/write?org=" + _org + "&bucket=" + _bucket, true);
```

File: InfluxDb.cpp (hunk 1 above)
Comment (tobiasschuerg) [reply InfluxDb.cpp]: Ah just saw you comment about `missing 'Token' header`.
Of course it makes sense to set the header for each request. 👍 

But then we can remove line 104.

**Comment by soudrug** [comment]
This will not work for ESP32.  There is not #include for WifiClientSecure.h and WifiClientSecure for ESP32 doesn't have ability to validate certificate with fingerprint.

File: InfluxDb.cpp (hunk 1 above)
Comment (cail) [reply InfluxDb.cpp]: agreed

File: InfluxDb.cpp (hunk 2 above)
Comment (cail) [reply InfluxDb.cpp]: sure

**Comment by cail** [comment]
> This will not work for ESP32. There is not #include for WifiClientSecure.h and WifiClientSecure for ESP32 doesn't have ability to validate certificate with fingerprint.

Hmm.. Then, this should be under #ifdef I believe?

**Comment by cail** [comment]
conditionally compiling out ESP32 for this feature

**Pull request merged**